src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -327,74 +288,6 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
     }
   }
 
-  /**
-   * If kafka offset was recently reset, we will skip adding any more records to buffer until we see
-   * a desired offset from kafka.
-   *
-   * <p>Desired Offset from Kafka = (offset persisted in snowflake + 1)
-   *
-   * <p>Check {link {@link TopicPartitionChannel#resetChannelMetadataAfterRecovery}} for reset logic
-   *
-   * @param kafkaSinkRecord Record to check for above condition only in case of failures
-   *     (isOffsetResetInKafka = true)
-   * @return true if this record can be skipped to add into buffer, false otherwise.
-   */
-  private boolean shouldIgnoreAddingRecordToBuffer(SinkRecord kafkaSinkRecord) {
-    if (this.isOffsetResetInKafka.get()) {
-      if ((kafkaSinkRecord.kafkaOffset() - offsetPersistedInSnowflake.get()) != 1L) {
-        // ignore
-        LOGGER.debug(
-            "Ignore adding offset:{} to buffer for channel:{} because we recently encountered"
```

Comment:
I would actually want to keep this debug statements. Can you put an else in above code and put a debug statement there?